stb_vorbis.c

```diff
@@ -553,7 +553,7 @@ enum STBVorbisError
 #include <string.h>
 #include <assert.h>
 #include <math.h>
-#if !(defined(__APPLE__) || defined(MACOSX) || defined(macintosh) || defined(Macintosh))
```

Comment:
I think Windows/MSVC (at least older versions) need malloc.h
Any recentish Unix-like systems shouldn't support using stdlib.h (and apparently more and more actually don't support malloc.h anymore). I checked the manpages of malloc(3) for Linux, the BSDs, OSX, Solaris, AIX, HP-UX.
QNX seems to use stdlib.h, too.
No idea about any proprietary systems like game console SDKs, I guess you have more experience with those :-)